From fa212bc0a29a7f8150cfbfcfdf553a23ba96df01 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Mon, 14 Oct 2024 19:35:20 +0200
Subject: [PATCH 1/2] reuse: Convert to REUSE.toml

The latest iteration of the spec deprecates .reuse/dep5 in favor
of REUSE.toml. Not a fan of the change (more in-your-face, makes
completing for README.md harder), but what do you do :shrug:
---
 .reuse/dep5 | 40 ----------------------------------------
 REUSE.toml  | 44 ++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 44 insertions(+), 40 deletions(-)
 delete mode 100644 .reuse/dep5
 create mode 100644 REUSE.toml

diff --git a/.reuse/dep5 b/.reuse/dep5
deleted file mode 100644
index 794fb392..00000000
--- a/.reuse/dep5
+++ /dev/null
@@ -1,40 +0,0 @@
-Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
-Upstream-Name: polari
-Upstream-Contact: Florian Müllner <fmuellner@gnome.org>
-Source: https://gitlab.gnome.org/GNOME/polari
-
-Files: AUTHORS NEWS README.md
-Copyright: No rights reserved
-License: CC0-1.0
-
-Files: *.service.in *.client *.desktop.in data/resources/networks.json
-Copyright: Florian Müllner <fmuellner@gnome.org>
-License: GPL-2.0-or-later
-
-# managed by translation teams
-Files: po/*.po help/*/*.po
-Copyright: GNOME Translation Teams <i18n@gnome.org>
-License: GPL-2.0-or-later
-
-# managed by translation teams
-Files: po/LINGUAS help/LINGUAS po/POTFILES.in
-Copyright: No rights reserved
-License: CC0-1.0
-
-# managed by docs team
-Files: help/*.page help/*.xml
-Copyright: GNOME documentation team <gnome-doc-list@gnome.org>
-License: CC-BY-SA-4.0
-
-#screenshots
-Files: data/metainfo/*.png help/*.png
-Copyright: No rights reserved
-License: CC0-1.0
-
-Files: flatpak/org.gnome.Polari.json
-Copyright: No rights reserved
-License: CC0-1.0
-
-Files: flatpak/*.patch
-Copyright: Florian Müllner <fmuellner@gnome.org>
-License: GPL-2.0-or-later
diff --git a/REUSE.toml b/REUSE.toml
new file mode 100644
index 00000000..bbbade87
--- /dev/null
+++ b/REUSE.toml
@@ -0,0 +1,44 @@
+version = 1
+SPDX-PackageName = "polari"
+SPDX-PackageSupplier = "Florian Müllner <fmuellner@gnome.org>"
+SPDX-PackageDownloadLocation = "https://gitlab.gnome.org/GNOME/polari"
+
+[[annotations]]
+path = ["AUTHORS", "NEWS", "README.md", "REUSE.toml"]
+SPDX-FileCopyrightText = "No rights reserved"
+SPDX-License-Identifier = "CC0-1.0"
+
+[[annotations]]
+path = ["**.service.in", "**.client", "**.desktop.in", "data/resources/networks.json"]
+SPDX-FileCopyrightText = "Florian Müllner <fmuellner@gnome.org>"
+SPDX-License-Identifier = "GPL-2.0-or-later"
+
+[[annotations]]
+path = ["po/**.po", "help/**/**.po"]
+SPDX-FileCopyrightText = "GNOME Translation Teams <i18n@gnome.org>"
+SPDX-License-Identifier = "GPL-2.0-or-later"
+
+[[annotations]]
+path = ["po/LINGUAS", "help/LINGUAS", "po/POTFILES.in"]
+SPDX-FileCopyrightText = "No rights reserved"
+SPDX-License-Identifier = "CC0-1.0"
+
+[[annotations]]
+path = ["help/**.page", "help/**.xml"]
+SPDX-FileCopyrightText = "GNOME documentation team <gnome-doc-list@gnome.org>"
+SPDX-License-Identifier = "CC-BY-SA-4.0"
+
+[[annotations]]
+path = ["data/metainfo/**.png", "help/**.png"]
+SPDX-FileCopyrightText = "No rights reserved"
+SPDX-License-Identifier = "CC0-1.0"
+
+[[annotations]]
+path = "flatpak/org.gnome.Polari.json"
+SPDX-FileCopyrightText = "No rights reserved"
+SPDX-License-Identifier = "CC0-1.0"
+
+[[annotations]]
+path = "flatpak/**.patch"
+SPDX-FileCopyrightText = "Florian Müllner <fmuellner@gnome.org>"
+SPDX-License-Identifier = "GPL-2.0-or-later"
-- 
GitLab


From 8f7331960baab8cfea029b39d9fd415e53267969 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Mon, 14 Oct 2024 19:56:58 +0200
Subject: [PATCH 2/2] reuse: Use REUSE.toml for icon licenses

We currently install the .license files along the icons, which
no longer passes flathub's linter:
https://docs.flathub.org/docs/for-app-authors/linter#non-svg-icon-in-scalable-folder

Avoid this by moving the license information for icons from
separate .license files into a common REUSE.toml file.
---
 REUSE.toml                                       |  2 +-
 data/icons/REUSE.toml                            | 16 ++++++++++++++++
 .../apps/org.gnome.Polari.Devel.svg.license      |  3 ---
 .../scalable/apps/org.gnome.Polari.svg.license   |  4 ----
 .../apps/org.gnome.Polari-symbolic.svg.license   |  3 ---
 5 files changed, 17 insertions(+), 11 deletions(-)
 create mode 100644 data/icons/REUSE.toml
 delete mode 100644 data/icons/hicolor/scalable/apps/org.gnome.Polari.Devel.svg.license
 delete mode 100644 data/icons/hicolor/scalable/apps/org.gnome.Polari.svg.license
 delete mode 100644 data/icons/hicolor/symbolic/apps/org.gnome.Polari-symbolic.svg.license

diff --git a/REUSE.toml b/REUSE.toml
index bbbade87..e2df50cf 100644
--- a/REUSE.toml
+++ b/REUSE.toml
@@ -4,7 +4,7 @@ SPDX-PackageSupplier = "Florian Müllner <fmuellner@gnome.org>"
 SPDX-PackageDownloadLocation = "https://gitlab.gnome.org/GNOME/polari"
 
 [[annotations]]
-path = ["AUTHORS", "NEWS", "README.md", "REUSE.toml"]
+path = ["AUTHORS", "NEWS", "README.md", "REUSE.toml", "data/icons/REUSE.toml"]
 SPDX-FileCopyrightText = "No rights reserved"
 SPDX-License-Identifier = "CC0-1.0"
 
diff --git a/data/icons/REUSE.toml b/data/icons/REUSE.toml
new file mode 100644
index 00000000..92264cf9
--- /dev/null
+++ b/data/icons/REUSE.toml
@@ -0,0 +1,16 @@
+version = 1
+
+[[annotations]]
+path = "hicolor/scalable/apps/org.gnome.Polari.svg"
+SPDX-FileCopyrightText = "2013 Sam Hewitt <hewittsamuel@gmail.com>, 2018 Tobias Bernard <tbernard@gnome.org>"
+SPDX-License-Identifier = "LGPL-2.0-or-later"
+
+[[annotations]]
+path = "hicolor/symbolic/apps/org.gnome.Polari-symbolic.svg"
+SPDX-FileCopyrightText = "2015 Jakub Steiner <jimmac@gmail.com>"
+SPDX-License-Identifier = "LGPL-2.0-or-later"
+
+[[annotations]]
+path = "hicolor/scalable/apps/org.gnome.Polari.Devel.svg"
+SPDX-FileCopyrightText = "2021 Tobias Bernard <tbernard@gnome.org>"
+SPDX-License-Identifier = "LGPL-2.0-or-later"
diff --git a/data/icons/hicolor/scalable/apps/org.gnome.Polari.Devel.svg.license b/data/icons/hicolor/scalable/apps/org.gnome.Polari.Devel.svg.license
deleted file mode 100644
index e878d1ad..00000000
--- a/data/icons/hicolor/scalable/apps/org.gnome.Polari.Devel.svg.license
+++ /dev/null
@@ -1,3 +0,0 @@
-SPDX-FileCopyrightText: 2021 Tobias Bernard <tbernard@gnome.org>
-
-SPDX-License-Identifier: LGPL-2.0-or-later
diff --git a/data/icons/hicolor/scalable/apps/org.gnome.Polari.svg.license b/data/icons/hicolor/scalable/apps/org.gnome.Polari.svg.license
deleted file mode 100644
index 6ca43afc..00000000
--- a/data/icons/hicolor/scalable/apps/org.gnome.Polari.svg.license
+++ /dev/null
@@ -1,4 +0,0 @@
-SPDX-FileCopyrightText: 2013 Sam Hewitt <hewittsamuel@gmail.com>
-SPDX-FileCopyrightText: 2018 Tobias Bernard <tbernard@gnome.org>
-
-SPDX-License-Identifier: LGPL-2.0-or-later
diff --git a/data/icons/hicolor/symbolic/apps/org.gnome.Polari-symbolic.svg.license b/data/icons/hicolor/symbolic/apps/org.gnome.Polari-symbolic.svg.license
deleted file mode 100644
index 57a22d4d..00000000
--- a/data/icons/hicolor/symbolic/apps/org.gnome.Polari-symbolic.svg.license
+++ /dev/null
@@ -1,3 +0,0 @@
-SPDX-FileCopyrightText: 2015 Jakub Steiner <jimmac@gmail.com>
-
-SPDX-License-Identifier: LGPL-2.0-or-later
-- 
GitLab

